api/server/httputils/errors.go

```diff
@@ -72,6 +72,52 @@ func GetHTTPErrorStatusCode(err error) int {
 	return statusCode
 }
 
+// FromStatusCode creates an errdef error, based on the provided status-code
+func FromStatusCode(err error, statusCode int) error {
```

Comment:
Did we mention we'd move this to errdefs?

Reply:
ach, yes, slipped my mind; let me work in that one